internal/handler/http/balance.go

```diff
@@ -0,0 +1,168 @@
+package handler
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	_ "github.com/jackc/pgx/v5/pgconn/ctxwatch"
+	"github.com/rookgm/gophermart/internal/models"
+	"net/http"
+	"time"
+)
+
+type BalanceService interface {
+	// GetBalance returns current user balance
+	GetBalance(ctx context.Context, userID uint64) (models.Balance, error)
+	// BalanceWithdrawal withdrawals of balance
+	BalanceWithdrawal(ctx context.Context, withdraw *models.Withdraw) (*models.Withdraw, error)
+	// GetWithdrawals returns user withdrawals
+	GetWithdrawals(ctx context.Context, userID uint64) ([]models.Withdraw, error)
+}
+
+// BalanceHandler represents HTTP handler for balance-related requests
+type BalanceHandler struct {
+	svc BalanceService
+}
+
+// NewBalanceHandler creates new BalanceHandler instance
+func NewBalanceHandler(svc BalanceService) *BalanceHandler {
+	return &BalanceHandler{svc: svc}
+}
+
+type balanceResponse struct {
+	Current   float64 `json:"current"`
+	Withdrawn float64 `json:"withdrawn"`
+}
+
+// GetUserBalance returns current user balance
+// 200 — успешная обработка запроса.
+// 401 — пользователь не авторизован.
+// 500 — внутренняя ошибка сервера.
+func (bh *BalanceHandler) GetUserBalance() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		authPayload, ok := getAuthPayload(r.Context(), authPayloadKey)
+		if authPayload == nil || !ok {
```

Comment:
Мне кажется стоит проверять `ok` в первую очередь, так как эта переменная для этого предназначена, а так как выражение `||` является "ленивым" если она будет проверена то последующие проверки уже не будут выполняться.
```suggestion
		if !ok || authPayload == nil {
```

Reply:
Согласен, поправлю.

Reply:
Поправил!